### PR TITLE
test: modify python test

### PR DESCRIPTION
**Pull request by wep21**
- modify library output path and add append_env to make python library available in tests.

### CODE REVIEW

**Review by audrow — COMMENTED** [review]
I'm not too knowledgable about these changes. @sloretz or @ivanpauno, would you mind taking a look?

File: cv_bridge/src/CMakeLists.txt
Comment (ivanpauno) [inline cv_bridge/src/CMakeLists.txt]: shouldn't you use `${PYTHON_INSTALL_DIR}/${PROJECT_NAME}/boost/` instead?

**Comment by ivanpauno** [comment]
@wep21 are the tests not working for you when you run them in what platform?
what's the error message when they fail?

File: cv_bridge/src/CMakeLists.txt
Comment (wep21) [reply cv_bridge/src/CMakeLists.txt]: I change to refer to the python library under install at https://github.com/ros-perception/vision_opencv/pull/444/commits/fe5d6eddd1d9bcb3a9f40fc9caa7395b7c7c7c9d.

**Comment by wep21** [comment]
@ivanpauno 

> are the tests not working for you when you run them in what platform?
> what's the error message when they fail?

I run the tests on Ubuntu 22.04. First of all, I removed `cv_bridge` installed under `/opt/ros`.
`sudo apt remove ros-humble-cv-bridge`

The error message said that `ModuleNotFoundError: No module named 'cv_bridge'`

**Comment by ijnek** [comment]
Thought I'd add in an instance of the error message from the buildfarm for reference:

```
03:17:09 2: ==================================== ERRORS ====================================
03:17:09 2: ________________ ERROR collecting cv_bridge/test/enumerants.py _________________
03:17:09 2: ImportError while importing test module '/tmp/ws/src/vision_opencv/cv_bridge/test/enumerants.py'.
03:17:09 2: Hint: make sure your test modules/packages have valid Python names.
03:17:09 2: Traceback:
03:17:09 2: /usr/lib/python3.10/importlib/__init__.py:126: in import_module
03:17:09 2:     return _bootstrap._gcd_import(name[level:], package, level)
03:17:09 2: /tmp/ws/src/vision_opencv/cv_bridge/test/enumerants.py:4: in <module>
03:17:09 2:     from cv_bridge import CvBridge, CvBridgeError, getCvType
03:17:09 2: E   ModuleNotFoundError: No module named 'cv_bridge'
03:17:09 2: -- generated xml file: /tmp/ws/test_results/cv_bridge/enumerants.py.xunit.xml --
03:17:09 2: =========================== short test summary info ============================
03:17:09 2: ERROR ../../../src/vision_opencv/cv_bridge/test/enumerants.py
03:17:09 2: !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
03:17:09 2: =============================== 1 error in 0.35s ===============================
03:17:09 2: -- run_test.py: return code 2
03:17:09 2: -- run_test.py: verify result file '/tmp/ws/test_results/cv_bridge/enumerants.py.xunit.xml'
03:17:09 2/4 Test #2: enumerants.py ....................***Failed    0.73 sec
```

**Comment by ivanpauno** [comment]
@wep21 I'm experiencing this issue in rolling as well, but it's not only for the tests.
If I source the workspace where I build `cv_bridge` from source and run

```bash
python3
> import cv_bridge
```

that fails.
I checked the install directories, and it looks that it's installing the python package in an unusual place (or one that colcon doesn't handle well).
We should be fixing that, instead of working the problem arround for the test cases.

**Comment by wep21** [comment]
@ivanpauno 
I found out that the root cause is `ament_python_install_package()` in sub directory.
The environment hook for python path disappears due to the scope. https://github.com/ament/ament_cmake/blob/65a3ad5f128e4b727187fa38c10af2fc6d4f9d53/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake#L80
debug output:
before
```
CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:77 (message):
  share/cv_bridge/environment/pythonpath.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_python/cmake/ament_cmake_python-extras.cmake:35 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_python/cmake/ament_python_install_package.cmake:38 (_ament_cmake_python_register_environment_hook)
  python/CMakeLists.txt:6 (ament_python_install_package)


CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:82 (message):
  ENV HOOKS share/cv_bridge/environment/pythonpath.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_python/cmake/ament_cmake_python-extras.cmake:35 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_python/cmake/ament_python_install_package.cmake:38 (_ament_cmake_python_register_environment_hook)
  python/CMakeLists.txt:6 (ament_python_install_package)


CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:77 (message):
  share/cv_bridge/environment/ament_prefix_path.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_cmake_environment_hooks_package_hook.cmake:20 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_package.cmake:66 (ament_execute_extensions)
  CMakeLists.txt:88 (ament_package)


CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:82 (message):
  ENV HOOKS share/cv_bridge/environment/ament_prefix_path.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_cmake_environment_hooks_package_hook.cmake:20 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_package.cmake:66 (ament_execute_extensions)
  CMakeLists.txt:88 (ament_package)


CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:77 (message):
  share/cv_bridge/environment/path.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_cmake_environment_hooks_package_hook.cmake:20 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_package.cmake:66 (ament_execute_extensions)
  CMakeLists.txt:88 (ament_package)


CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:82 (message):
  ENV HOOKS
  share/cv_bridge/environment/ament_prefix_path.sh;share/cv_bridge/environment/path.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_cmake_environment_hooks_package_hook.cmake:20 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_package.cmake:66 (ament_execute_extensions)
  CMakeLists.txt:88 (ament_package)
```
after
```
CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:77 (message):
  share/cv_bridge/environment/pythonpath.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_python/cmake/ament_cmake_python-extras.cmake:35 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_python/cmake/ament_python_install_package.cmake:38 (_ament_cmake_python_register_environment_hook)
  CMakeLists.txt:62 (ament_python_install_package)


CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:82 (message):
  ENV HOOKS share/cv_bridge/environment/pythonpath.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_python/cmake/ament_cmake_python-extras.cmake:35 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_python/cmake/ament_python_install_package.cmake:38 (_ament_cmake_python_register_environment_hook)
  CMakeLists.txt:62 (ament_python_install_package)


CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:77 (message):
  share/cv_bridge/environment/ament_prefix_path.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_cmake_environment_hooks_package_hook.cmake:20 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_package.cmake:66 (ament_execute_extensions)
  CMakeLists.txt:95 (ament_package)


CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:82 (message):
  ENV HOOKS
  share/cv_bridge/environment/pythonpath.sh;share/cv_bridge/environment/ament_prefix_path.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_cmake_environment_hooks_package_hook.cmake:20 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_package.cmake:66 (ament_execute_extensions)
  CMakeLists.txt:95 (ament_package)


CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:77 (message):
  share/cv_bridge/environment/path.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_cmake_environment_hooks_package_hook.cmake:20 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_package.cmake:66 (ament_execute_extensions)
  CMakeLists.txt:95 (ament_package)


CMake Warning at /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake:82 (message):
  ENV HOOKS
  share/cv_bridge/environment/pythonpath.sh;share/cv_bridge/environment/ament_prefix_path.sh;share/cv_bridge/environment/path.sh
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_core/cmake/environment_hooks/ament_cmake_environment_hooks_package_hook.cmake:20 (ament_environment_hooks)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_package.cmake:66 (ament_execute_extensions)
  CMakeLists.txt:95 (ament_package)
```
I moved CMakeLists.txt contents under python directory into CMakeLists.txt in the root directory.  https://github.com/ros-perception/vision_opencv/pull/444/commits/dad85254abfe380b47f03aaaaa813880d666858d

File: cv_bridge/test/CMakeLists.txt
Comment (ivanpauno) [inline cv_bridge/test/CMakeLists.txt]: double checking, are these changes needed after fixing the `ament_python_install_package()`?
IIUC, colcon should figure it out now

File: cv_bridge/CMakeLists.txt
Comment (ivanpauno) [inline cv_bridge/CMakeLists.txt]: There's nothing to configure in __init__.py, as it's an empty file.
There's also an __init__.py [here](https://github.com/ros-perception/vision_opencv/blob/ros2/cv_bridge/python/cv_bridge/__init__.py), so I think we can just remove this.

```suggestion
```

**Comment by ivanpauno** [comment]
> I found out that the root cause is ament_python_install_package() in sub directory.
The environment hook for python path disappears due to the scope. https://github.com/ament/ament_cmake/blob/65a3ad5f128e4b727187fa38c10af2fc6d4f9d53/ament_cmake_core/cmake/environment_hooks/ament_environment_hooks.cmake#L80

Thanks @wep21 !!!

File: cv_bridge/test/CMakeLists.txt
Comment (wep21) [reply cv_bridge/test/CMakeLists.txt]: I confirmed the tests passed without those changes. I dropped them.

File: cv_bridge/CMakeLists.txt
Comment (wep21) [reply cv_bridge/CMakeLists.txt]: I removed __init__.py in https://github.com/ros-perception/vision_opencv/pull/444/commits/fcdd52b29bf45b9c6aa1905c415858f652d0d574.

**Review by ivanpauno — APPROVED** [review]
Code looks good!
I have also tested locally and it's working fine.

**Comment by ivanpauno** [comment]
@audrow do you have merge access here?
In that case, could you merge?

If you do that, I would also close https://github.com/ros-perception/vision_opencv/pull/446, which is doing (more or less) the same thing.

**Review by audrow — APPROVED** [review]

**Comment by ijnek** [comment]
With this PR merged, may I ask one of the maintainers of the repo to make a new release into rolling please?

**Comment by ijnek** [comment]
@audrow or @ivanpauno With this PR merged, may I ask one of the maintainers of the repo to make a new release into rolling please?

**Comment by ivanpauno** [comment]
@audrow can you make a release to rolling?
I don't have write access here.

@wep21 @ijnek was this affecting `humble` as well? or only rolling?

**Comment by ijnek** [comment]
Yes, the [Humble dev CIs](https://build.ros2.org/job/Hdev__vision_opencv__ubuntu_jammy_amd64/) started passing after these changes were merged so a humble release should be made too.

(The humble dev branch points to the ros2 branch)

**Pull request merged**
